File: FestivalApp-SPA/src/app/_services/festival.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Festival } from '../_models/festival';
import { environment } from 'src/environments/environment';
import { User } from '../_models/user';
import { Rental } from '../_models/rental';

@Injectable({
  providedIn: 'root',
})
export class FestivalService {
  baseUrl = environment.apiUrl;

  constructor(private http: HttpClient) {}

  getFestivals(userId: number) {
    return this.http.get<Festival[]>(this.baseUrl + 'festivals/' + userId);
  }

  getRentals() {
    return this.http.get<Rental[]>(this.baseUrl + 'rentals');
  }

  getUser(id: number) {
    return this.http.get<User>(this.baseUrl + 'users/' + id);
  }

  updateUser(id: number, user: User) {
    return this.http.put(this.baseUrl + 'users/' + id, user);
  }

  likeFestival(festivalId: number, userId: number) {
    return this.http.post(this.baseUrl + 'festivals/' + festivalId + '/like/' + userId, {});
  }
}

File: FestivalApp/FestivalApp.Core/Services/FestivalService.cs
using FestivalApp.Core.Interfaces;
using FestivalApp.Domain.Models;

namespace FestivalApp.Core.Services
{
    internal class FestivalService : IFestivalService
    {
        private readonly IFestivalRepository _festivalRepository;

        public FestivalService(IFestivalRepository festivalRepository)
        {
            _festivalRepository = festivalRepository;
        }

        public Festival AddFestival(Festival festival)
        {
            throw new NotImplementedException();
        }

        public List<Festival> GetFestivals(int userId)
        {
            throw new NotImplementedException();
        }

        public void LikeFestival(int festivalId, int userId)
        {
            throw new NotImplementedException();
        }
    }
}
